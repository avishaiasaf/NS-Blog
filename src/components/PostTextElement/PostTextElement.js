import React from 'react';

function NewlineText(props) {
    const text = props.text;
    return text.split('\n').map(str => {
        if(str.match(/\*([^*]*)\*/)){
            console.log('match');
            return <p className='f3 lh-copy w-100 tl'><strong>{str.replace(/\*/g,'')}</strong></p>
        }else{
            return <p className='f3 lh-copy w-100 tl'>{str}</p>
        }
    });
  }

const PostTextElement = ( {text} )=>{
    return (
        <div>
           <NewlineText text={text}  />
            {/* <p className='f3 lh-copy w-100 tl'>{text}</p> */}
        </div>
    );
}

export default PostTextElement;