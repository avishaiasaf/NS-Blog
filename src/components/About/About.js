import React from 'react';
// import about_me from '../../images/about-me.jpg';
import PostTextElement from '../PostTextElement/PostTextElement';

const About = ()=>{
    return (
        <div className='center w-60' >
            <h1>About</h1>
            {/*<img src={ about_me } alt='About Pic' align={'left'} width="20%" style={{paddingRight: '30px'}} />*/}
            <PostTextElement text={`

            `} />
        </div>
    );
}

export default About;