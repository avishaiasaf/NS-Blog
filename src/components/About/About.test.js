// import React from "react";
// import { render, unmountComponentAtNode } from "react-dom";
// import { act } from "react-dom/test-utils";
// import About from "./About";
//
// let container = null;
// beforeEach(() => {
//     // setup a DOM element as a render target
//     container = document.createElement("div");
//     document.body.appendChild(container);
// });
//
// afterEach(() => {
//     // cleanup on exiting
//     unmountComponentAtNode(container);
//     container.remove();
//     container = null;
// });
//
// it("renders with or without a name", () => {
//     act(() => {
//         render(<About />, container);
//     });
//     expect(container.textContent).toBe("About            ");
// });

import React from 'react';
import renderer from 'react-test-renderer';
import About from "./About";
import about_me from "../../images/about-me.jpg";
import PostTextElement from "../PostTextElement/PostTextElement";

it('renders correctly', () => {
    const tree = renderer
        .create(
            <div className='center w-60' >
                <h1>About</h1>
                <img src={ about_me } alt='About Pic' align={'left'} width="20%" style={{paddingRight: '30px'}} />
                <PostTextElement text={`
                
            `} />
            </div>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});