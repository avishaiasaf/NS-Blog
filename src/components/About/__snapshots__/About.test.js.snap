// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="center w-60"
>
  <h1>
    About
  </h1>
  <img
    align="left"
    alt="About Pic"
    src="about-me.jpg"
    style={
      Object {
        "paddingRight": "30px",
      }
    }
    width="20%"
  />
  <div>
    <p
      className="f5 lh-copy w-100 tl"
    >
      
    </p>
    <p
      className="f5 lh-copy w-100 tl"
    >
                      
    </p>
    <p
      className="f5 lh-copy w-100 tl"
    >
                  
    </p>
  </div>
</div>
`;
